rootex/core/renderer/post_processor.cpp

```diff
@@ -153,6 +153,55 @@ class MonochromePostProcess : public PostProcess
 	}
 };
 
+Variant CustomPostProcess::loadRTVAndSRV(const Event* event)
+{
+	RenderingDevice::GetSingleton()->createRTVAndSRV(m_CacheRTV, m_CacheSRV);
+	return true;
+}
+
+CustomPostProcess::CustomPostProcess(String path)
+{
+	postProcessPath = path;
+
+	m_Binder.bind(RootexEvents::WindowResized, this, &CustomPostProcess::loadRTVAndSRV);
+
+	loadRTVAndSRV(nullptr);
+
+	m_BasicPostProcess.reset(new DirectX::BasicPostProcess(RenderingDevice::GetSingleton()->getDevice()));
+}
+
+void CustomPostProcess::draw(CameraComponent* camera, ID3D11ShaderResourceView*& nextSource)
+{
+	const PostProcessingDetails& postProcessingDetails = camera->getPostProcessingDetails();
+	for (auto&& details : postProcessingDetails.m_CustomPostProcessing)
+	{
+		if (details.path == postProcessPath && details.isSelected == true)
+		{
+			RenderingDevice::GetSingleton()->unbindSRVs();
+			RenderingDevice::GetSingleton()->setRTV(m_CacheRTV.Get());
+
+			Microsoft::WRL::ComPtr<ID3D11ShaderResourceView> texture = nextSource;
+			ID3D11ShaderResourceView* textures[1] = { texture.Get() };
+			RenderingDevice::GetSingleton()->setPSSRV(0, 1, textures);
+
+			BufferFormat bufferFormat;
+			bufferFormat.push(VertexBufferElement::Type::FloatFloatFloat, "POSITION", D3D11_INPUT_PER_VERTEX_DATA, 0, false, 0);
+			bufferFormat.push(VertexBufferElement::Type::FloatFloatFloat, "NORMAL", D3D11_INPUT_PER_VERTEX_DATA, 0, false, 0);
+			bufferFormat.push(VertexBufferElement::Type::FloatFloat, "TEXCOORD", D3D11_INPUT_PER_VERTEX_DATA, 0, false, 0);
+			bufferFormat.push(VertexBufferElement::Type::FloatFloatFloat, "TANGENT", D3D11_INPUT_PER_VERTEX_DATA, 0, false, 0);
+
+			Ptr<Shader> shader(new Shader("rootex/core/renderer/shaders/custom_post_processing_vertex_shader.hlsl", postProcessPath, bufferFormat));
```

Comment:
Creating a new `Shader` object on every draw call O_O
It is needed to be created only once and save it in the members